### PR TITLE
Refactor/active app

### PR DESCRIPTION
**Pull request by Leptopoda**

### CODE REVIEW

File: packages/neon/neon/lib/src/blocs/apps.dart [hunk 1]
```diff
@@ -13,7 +13,7 @@ abstract class AppsBlocStates {
 
   BehaviorSubject<Result<NotificationsAppInterface?>> get notificationsAppImplementation;
 
-  BehaviorSubject<String?> get activeAppID;
+  BehaviorSubject<AppImplementation> get activeApp;
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/blocs/apps.dart:16]: Should have been done this way from the start :disappointed:

File: packages/neon/neon/lib/src/blocs/apps.dart [hunk 2]
```diff
@@ -3,7 +3,7 @@ part of 'blocs.dart';
 typedef NextcloudApp = NextcloudCoreNavigationApps_Ocs_Data;
 
 abstract class AppsBlocEvents {
-  void setActiveApp(final String? appID);
+  void setActiveApp(final String appID);
```
Comment (provokateurin) [inline packages/neon/neon/lib/src/blocs/apps.dart:6]: Why is this required?

Comment (Leptopoda) [reply packages/neon/neon/lib/src/blocs/apps.dart:6]: 1. I don't see why we should allow it. I mean how would we set the active app to nothing? If we ever need this we could add something like `void clearActiveApp();` .

2. in the case null was passed we weren't doing anything and just returning (plus maybe wasting some time in the await depending on the state of the app). So it didn't really do anything.

Comment (provokateurin) [reply packages/neon/neon/lib/src/blocs/apps.dart:6]: I think if you had no compatible apps this method was triggered with a null value, but I guess that was a bug from the start. Please make sure this works when no apps are supported.

Comment (Leptopoda) [reply packages/neon/neon/lib/src/blocs/apps.dart:6]: Oh this is indeed something I forgot;
I just forced the nullcheck as I thought initialApp will be set to a value before calling this but having a closer look it can indeed be null.

I'll just check it before calling setActiveApp like:

```dart
if (!activeApp.hasValue && initialApp != null) {
  await setActiveApp(initialApp!);
}
```

**Review by provokateurin — APPROVED** [review]

**Comment by Leptopoda** [comment]
Did the CI get flaky again? :(

**Comment by provokateurin** [comment]
The tests are fine. I've been seeing increased network failures during requests, but I don't know why. I think we could enabled retries on failed tests, that should make it more stable.
